packages/buidler-core/src/builtin-tasks/jsonrpc.ts

```diff
@@ -0,0 +1,66 @@
+import debug from "debug";
+
+import {
+  JsonRpcServer,
+  JsonRpcServerConfig
+} from "../internal/buidler-evm/jsonrpc/server";
+import { BUIDLEREVM_NETWORK_NAME } from "../internal/constants";
+import { task } from "../internal/core/config/config-env";
+import { BuidlerError } from "../internal/core/errors";
+import { ERRORS } from "../internal/core/errors-list";
+import { createProvider } from "../internal/core/providers/construction";
+import { lazyObject } from "../internal/util/lazy";
+import { EthereumProvider, ResolvedBuidlerConfig } from "../types";
+
+import { TASK_JSONRPC } from "./task-names";
+
+const log = debug("buidler:core:tasks:jsonrpc");
+
+function _createBuidlerRuntimeEnvironment(
+  config: ResolvedBuidlerConfig
+): EthereumProvider {
+  log("Creating BuidlerRuntimeEnvironment");
+
+  const networkName = BUIDLEREVM_NETWORK_NAME;
+  const networkConfig = config.networks[networkName];
+
+  return lazyObject(() => {
+    log(`Creating buidlerevm provider for JSON-RPC sever`);
+    return createProvider(
+      networkName,
+      networkConfig,
+      config.solc.version,
+      config.paths
+    );
+  });
+}
+
+export default function() {
+  task(TASK_JSONRPC, "Starts a buidler JSON-RPC server").setAction(
+    async (_, { config }) => {
+      try {
```

Comment:
What about adding some validation here that prevents users from using this task with a different network?

Something like:

```ts
if (config.network.name !== BUIDLEREVM_NETWORK_NAME) { 
   throw ...
```

Otherwise using this task with another network would be confusing.

Reply:
My thinking was not to rely on the network provided by the user and always bootstrap a new BuidlerEVM to use as the provider.

The user experience I was thinking to support (which wouldn't be as clean with the exception you mention) is the following:
- I am working on a dApp, so I am actively working with the jsonrpc
- jsonrpc is the same as buidlerevm  for the rest of the tooling (its the same evm in the end) and doesn't make much difference which one I choose.
- I set the default network to be http://localhost:8545, so I don't have to provide the --network all the time
- I run the jsonrpc just by using `buidler jsonrpc`

In this scenario, if we were to throw the exception you mention, each time you wanted to start your json-rpc server, you would have to specify --network=buidlerevm

Reply:
Makes sense! I guess a better option would be:

```ts
if (bre.buidlerArguments.network !== undefined) {
  throw new BuidlerErrorIndicatingThatTheNetworkParamDoesNotWorkHere();
}
```

Reply:
I think this is resolved now.